### PR TITLE
Add Scala Native support

### PR DESCRIPTION
**Pull request by plokhotnyuk**
Currently, using a SNAPSHOT version of cldr-api with SN support I'm getting the following compiler error:
```
[error] Error while emitting locales/cldr/data/_gd$
[error] Method too large: locales/cldr/data/_gd$.<init> ()V
[error] one error found
[error] (localesFullCurrenciesDb / Compile / compileIncremental) Compilation failed
```

### CODE REVIEW

**Comment by cquiroz** [comment]
Depending on the scala version and the cldr version there maybe too much data on the generated code. The solution is to change the generator and split the data in several classes. That is done on `sbt-tzdb` but I haven't had time to do it here

For 99% of user this is probably not an issue as you want to heavily filter the locales you want to include but the all locales modules needs it

File: build.sbt [hunk 1]
```diff
@@ -2,14 +2,14 @@ import sbtcrossproject.CrossPlugin.autoImport.{ CrossType, crossProject }
...
-lazy val cldrApiVersion = "2.5.0"
+lazy val cldrApiVersion = "2.5.1"
 
 Global / onChangedBuildSource := ReloadOnSourceChanges
 
 resolvers in Global += Resolver.sonatypeRepo("public")
 
-ThisBuild / scalaVersion := "2.13.3"
-ThisBuild / crossScalaVersions := Seq("2.11.12", "2.12.12", "2.13.3", "3.0.0-RC2", "3.0.0-RC3")
+ThisBuild / scalaVersion := "2.13.5"
```
Comment (cquiroz) [inline build.sbt:11]: Keep it on `2.13.3` to avoid the `method too large` bug

Comment (plokhotnyuk) [reply build.sbt]: Done

**Comment by cquiroz** [comment]
I released sbt-locales as 2.6.0. Could you update?

**Comment by cquiroz** [comment]
It seems there are some unformatted files on the native side.
Could you also remove `sbt-dotty` from the list of plugins?

**Comment by plokhotnyuk** [comment]
> It seems there are some unformatted files on the native side.
> Could you also remove `sbt-dotty` from the list of plugins?

I've fixed formatting and removed `sbt-dotty`, but the `testSuite` project doesn't compile for SN yet.

**Comment by cquiroz** [comment]
Could you re run `githubWorkflowGenerate`

**Comment by plokhotnyuk** [comment]
> Could you re run `githubWorkflowGenerate`

Re ran and force pushed regenerated `ci.yml`.

**Comment by avdv** [comment]
Hi.

I didn't realize someone is working on #38 so I started looking into this yesterday...

First, I also upgraded Scala 2.13 to 2.13.4 because scala-native supports 2.13.4, 2.12.13 and 2.11.12.
 
The only stopper I found (until now) is that there is an initialization cycle in `DefaultLocale.platformLocale -> Locale object init -> DefaultLocale.platformLocale` which leads to this error when trying to call `DefaultLocale.platformLocale`:
```
scala.UninitializedFieldError: Uninitialized field: /home/claudio/github/scala-java-locales/core/shared/src/main/scala/locales/LocalesDb.scala: 25
	at java.lang.Throwable.fillInStackTrace(Unknown Source)
	at locales.DefaultLocale$.platformLocale(Unknown Source)
	at java.util.Locale$.<init>(Unknown Source)
	at <none>._SM17java.util.Locale$G4load(Unknown Source)
	at java.util.Locale$LocaleBuilder.language(Unknown Source)
	at java.util.Locale$Builder.setLanguage(Unknown Source)
	at locales.cldr.LDML.toLocale(Unknown Source)
	at locales.cldr.LDML.languageTag(Unknown Source)
	at locales.cldr.fallback.LocalesProvider$.$anonfun$ldmls$1(Unknown Source)
	at locales.cldr.fallback.LocalesProvider$$$Lambda$1.apply(Unknown Source)
	at scala.collection.ArrayOps$.map$extension(Unknown Source)
	at locales.cldr.fallback.LocalesProvider$.ldmls(Unknown Source)
	at locales.LocalesDb$.<init>(Unknown Source)
	at <none>._SM18locales.LocalesDb$G4load(Unknown Source)
	at locales.DefaultLocale$.platformLocale(Unknown Source)
	at Main$.main(Unknown Source)
	at <none>.main(Unknown Source)
	at <none>.__libc_start_main(Unknown Source)
	at <none>._start(Unknown Source)
```
I was able to break the cycle by calling `Locale.setDefault(Locale.ENGLISH)` in `DefaultLocale.platformLocale` explicitly.

Then, my initial thought was that we need a proper way to determine the platform locale for the native environment. That's why I added the `setlocale` C library function and I am calling `setlocale(LC_ALL, "")` to initialize the locale; and then `setlocale(LC_ALL, null)` to query the locale information. 

I suppose this is what the JVM does too, I tested it with JNA simply calling the latter method and it returned the correct information (`en_US.UTF-8`).

Alternatively, we could simply inspect the `LC_ALL` or `LANG` variables to determine the platform locale, in order to avoid setting the locale by default. WDYT?

**Comment by cquiroz** [comment]
The JVM and scala js expect to have a default locale defined, many methods aren't specified if no locale is present.
the scala.js tries to find what's the locale from the browser environment and if all fails falls back to english

I think scala native should do the same, find what the locale is (I can't really say which method is best to do so) and if it fails default to english

**Comment by lolgab** [comment]
This can now be closed since it's deprecated by https://github.com/cquiroz/scala-java-locales/pull/307

**Comment by er1c** [comment]
Closed per https://github.com/cquiroz/scala-java-locales/pull/273#issuecomment-956811309

**Pull request closed without merging**